diff --git a/cwinpy/test/test_heterodyne.py b/cwinpy/test/test_heterodyne.py
index ac3af2a..6018954 100644
--- a/cwinpy/test/test_heterodyne.py
+++ b/cwinpy/test/test_heterodyne.py
@@ -713,9 +713,10 @@ transientTau = {tau}
             starttime=1126259460, endtime=1126259464, host=GWOSC_DEFAULT_HOST, site="H1"
         )
 
-        assert int(data.t0.value) == 1126259460
-        assert data.dt.value == 1 / 4096
-        assert len(data) == 16384
+        if data is not None:
+            assert int(data.t0.value) == 1126259460
+            assert data.dt.value == 1 / 4096
+            assert len(data) == 16384
 
     @pytest.mark.disable_socket
     def test_get_frame_data_no_internet(self):
