From bf44974065d7fd526a115f2e2a910423eb3b0765 Mon Sep 17 00:00:00 2001
From: "duncan.macleod" <duncan.macleod@ligo.org>
Date: Wed, 6 Dec 2023 14:24:01 +0000
Subject: [PATCH] test: mark all test_heterodyne_xxx as slow

---
 cwinpy/test/test_roq.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cwinpy/test/test_roq.py b/cwinpy/test/test_roq.py
index 3ff305b..2a879b9 100644
--- a/cwinpy/test/test_roq.py
+++ b/cwinpy/test/test_roq.py
@@ -581,6 +581,7 @@ phi0 = {phi0}
         shutil.rmtree(cls.fakepardir)
         shutil.rmtree(cls.fakedatadir)
 
+    @pytest.mark.slow
     def test_heterodyne_exact(self):
         """
         Heterodyne with the exact phase parameters
@@ -649,6 +650,7 @@ phi0 = {phi0}
                 gridroq.marginalize_ln_likelihood(not_parameters=par),
             )
 
+    @pytest.mark.slow
     def test_heterodyne_offset(self):
         """
         Heterodyne with offset phase parameters
@@ -893,6 +895,7 @@ phi0 = {phi0}
         shutil.rmtree(cls.fakepardir)
         shutil.rmtree(cls.fakedatadir)
 
+    @pytest.mark.slow
     def test_heterodyne_exact(self):
         """
         Heterodyne with the exact binary parameters
@@ -960,6 +963,7 @@ phi0 = {phi0}
                 gridroq.marginalize_ln_likelihood(not_parameters=par),
             )
 
+    @pytest.mark.slow
     def test_heterodyne_offset(self):
         """
         Heterodyne with offset binary parameters
-- 
GitLab

