From feab8ac298906508c41e01b738cf261bb733c306 Mon Sep 17 00:00:00 2001
From: Matthew Pitkin <matthew.pitkin@ligo.org>
Date: Mon, 1 Dec 2025 20:59:07 +0000
Subject: [PATCH] test_signal.py: add pytest.mark.xfail to tempo2 tests

---
 cwinpy/test/test_signal.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cwinpy/test/test_signal.py b/cwinpy/test/test_signal.py
index b012dca..d9312e2 100644
--- a/cwinpy/test/test_signal.py
+++ b/cwinpy/test/test_signal.py
@@ -346,6 +346,9 @@ class TestPhaseOnly:
         assert maxdfpole < 0.2 * dfmax
         assert maxdfpole < maxdfplane
 
+    @pytest.mark.xfail(
+        raises=(ValueError, UnicodeDecodeError), reason="libstempo issue"
+    )
     def test_tempo2(self):
         """
         Test the phase only when calculated with Tempo2 vs LAL.
@@ -520,6 +523,9 @@ phi0 = {phi0}
         shutil.rmtree(cls.fakepardir)
         shutil.rmtree(cls.fakedatadir)
 
+    @pytest.mark.xfail(
+        raises=(ValueError, UnicodeDecodeError), reason="libstempo issue"
+    )
     def test_ref_freq_heterodyne(self):
         segments = [
             (self.fakedatastart, self.fakedatastart + self.fakedataduration),
-- 
2.43.0

